frontend/.eslintrc

```
@@ -0,0 +1,48 @@
+{
+  "env": {
+    "browser": true,
+    "es2021": true
+  },
+
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": "latest",
+    "sourceType": "module"
+  },
+
+  "extends": [
+    "plugin:@typescript-eslint/recommended",
+    "airbnb",
+    "airbnb/hooks",
+    "prettier"
+  ],
+  "plugins": [
+    "@typescript-eslint",
+    "react",
+    "react-hooks",
+    "import",
+    "jsx-a11y"
+  ],
+
+  "globals": {
+    "JSX": "readonly"
+  },
+  "rules": {
+    // enable tsx extensions
+    "react/jsx-filename-extension": [1, { "extensions": [".tsx"] }],
+
+    // disable import resolving to sync with ts
+    "import/extensions": "off",
+    "import/no-unresolved": "off",
+
+    // additional ts settings for strictness
+    "@typescript-eslint/consistent-type-exports": "error",
+    "@typescript-eslint/consistent-type-imports": "error",
+    "@typescript-eslint/no-require-imports": "error",
+    "@typescript-eslint/promise-function-async": "error",
+    "default-param-last": "off",
+    "@typescript-eslint/default-param-last": "error",
+    "no-magic-numbers": "off",
+    "@typescript-eslint/no-magic-numbers": "error"
```

```suggestion
    "@typescript-eslint/no-magic-numbers": "off"
```  
  
This setting is causing a lot of unneeded linting errors - can probably just be removed.